File: WidgetsDemo/ImagesandRadioGroupsDemo/app/src/main/java/com/example/imagesandradiogroupsdemo/MainActivity.java
package com.example.imagesandradiogroupsdemo;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.RadioGroup;

import androidx.appcompat.app.AppCompatActivity;

import static android.widget.Toast.LENGTH_SHORT;
import static android.widget.Toast.makeText;
import static com.example.imagesandradiogroupsdemo.R.drawable.icon_sample;
import static com.example.imagesandradiogroupsdemo.R.id.imageView2;
import static com.example.imagesandradiogroupsdemo.R.id.radio;
import static com.example.imagesandradiogroupsdemo.R.id.radioButton;
import static com.example.imagesandradiogroupsdemo.R.id.radioButton2;
import static com.example.imagesandradiogroupsdemo.R.layout.activity_main;

public class MainActivity extends AppCompatActivity {
	ImageView imageView;
	RadioGroup radioGroup;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(activity_main);
		imageView = findViewById(imageView2);
		imageView.setImageResource(icon_sample);
		System.out.println(imageView.getId() + "\n" + icon_sample + "\n" + imageView2);
		radioGroup = findViewById(radio);
		radioGroup.setOnCheckedChangeListener((group, checkedId) -> {
			if (checkedId == radioButton) {
//					do something
			} else if (checkedId == radioButton2)
				makeText(MainActivity.this, "Hello", LENGTH_SHORT).show();
		});
	}
}

File: RandomTesting/app/src/main/java/com/example/randomtesting/MainActivity.kt
package com.example.randomtesting

import android.graphics.Bitmap
import android.graphics.BitmapFactory.decodeStream
import android.os.AsyncTask
import android.os.Bundle
import android.util.Log.e
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.example.randomtesting.R.id.imageView
import com.example.randomtesting.R.layout.activity_main
import java.net.URL


class MainActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(activity_main)
		DownloadImageTask(findViewById(imageView), "https://orteil.dashnet.org/cookieclicker/img/buildings.png").execute()
	}

	private class DownloadImageTask(var bmImage: ImageView, var url: String) : AsyncTask<String?, Void?, Bitmap?>() {
		protected override fun doInBackground(vararg params: String?): Bitmap? {
			var mIcon11: Bitmap? = null
			try {
				mIcon11 = decodeStream(URL(url).openStream())
			} catch (e: Exception) {
				e.message?.let { e("Error", it) }
				e.printStackTrace()
			}
			return mIcon11
		}

		override fun onPostExecute(result: Bitmap?) = bmImage.setImageBitmap(result)
	}
}